packages/trie/test/trie/prune.spec.ts

```diff
@@ -52,8 +52,48 @@ tape('Pruned trie tests', function (tester) {
     for (let i = 0; i < keys.length; i++) {
       //console.log(await trie.get(Buffer.from(keys[i], 'hex')))
     }
+    st.end()
+  })
+
+  // TODO
+
+  it('should prune when keys are updated or deleted', async (st) => {
+    const trie = new Trie({ pruneTrie: true })
+    const keys: string[] = []
+    for (let i = 0; i < 100; i++) {
+      keys.push(i.toString(16))
+    }
+    const values: string[] = []
+    for (let i = 0; i < 1000; i++) {
+      let val = Math.floor(Math.random() * 16384)
+      while (values.includes(val.toString(16))) {
+        val = Math.floor(Math.random() * 16384)
+      }
+      values.push(val.toString(16))
+    }
+    for (let i = 0; i < keys.length; i++) {
+      const idx = i < 100 ? i : Math.floor(Math.random() * 100)
+      const key = keys[idx]
+      await trie.put(Buffer.from(key), Buffer.from(values[i]))
+    }
+    st.equals((<any>trie.db)._database.size, 20, 'DB size correct')
```

Comment:
How come we know beforehand how many entries are in the DB if the test is random..?

Reply:
This makes absolutely no sense to me, if I run the test 1000 times it always passes!?